src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -53,12 +56,360 @@ uint32_t SystemRequest::index = 0;
+                    error_prefix_ << "QueryApps response is not array.");
+      return false;
+    }
+    bool result = true;
+    const std::size_t arr_size(obj_array->size());
+    for (std::size_t idx = 0; idx < arr_size; ++idx) {
+      const smart_objects::SmartObject& app_data = (*obj_array)[idx];
+
+      if (!app_data.isValid()) {
+        LOG4CXX_WARN(logger_,
+                     error_prefix_ << "Wrong application data in json file.");
+        result = false;
+        continue;
+      }
+
+      if (!ValidateAppIdAndAppName(app_data)) {
+        result = false;
+      }
+
+      // Verify os and dependent languages data
+      std::string os_type;
+      if (app_data.keyExists(json::ios)) {
+        os_type = json::ios;
+        if (!app_data[os_type].keyExists(json::urlScheme)) {
+          LOG4CXX_WARN(logger_,
+                       error_prefix_ << "Can't find URL scheme in json file.");
+          result = false;
+        } else {
+          if (app_data[os_type][json::urlScheme].asString().length() >
+              max_url_scheme_lengh_) {
+            LOG4CXX_WARN(
+                logger_,
+                error_prefix_
+                    << "An urlscheme length exceeds maximum allowed ["
+                    << app_data[os_type][json::urlScheme].asString().length()
+                    << "]>[" << max_url_scheme_lengh_ << "]");
+            result = false;
+          }
+        }
+      }
+      if (os_type.empty()) {
+        if (app_data.keyExists(json::android)) {
+          os_type = json::android;
+          if (!app_data[os_type].keyExists(json::packageName)) {
+            LOG4CXX_WARN(logger_,
+                         error_prefix_
+                             << "Can't find package name in json file.");
+            result = false;
+          } else {
+            if (app_data[json::android][json::packageName].asString().length() >
+                max_package_name_length_) {
+              LOG4CXX_WARN(logger_,
+                           error_prefix_
+                               << "Package name length ["
+                               << app_data[json::android][json::packageName]
+                                      .asString()
+                                      .length() << "] exceeds max length ["
+                               << max_package_name_length_ << "]in json file.");
+              result = false;
+            }
+          }
+        }
+      }
+
+      if (os_type.empty()) {
+        LOG4CXX_WARN(logger_,
+                     error_prefix_
+                         << "Can't find mobile OS type in json file.");
+        result = false;
+        continue;
+      }
+
+      // Languages verification
+      if (!app_data[os_type].keyExists(json::languages)) {
+        LOG4CXX_WARN(logger_, error_prefix_ << "\"languages\" not exists");
+        result = false;
+        continue;
+      }
+      if (!ValidateLanguages(app_data[os_type][json::languages]))
+        result = false;
```

Comment:
no, it'll brake logic cause result is cumulative after all verifications, for example ValidateLanguages() may return true for some data and produce error.